### PR TITLE
ci: observe NUM_CLUSTERS env var for e2e

### PR DESCRIPTION
**Pull request by sdowell**

### CODE REVIEW

File: e2e/flags.go [hunk 1]
```diff
@@ -90,7 +90,7 @@ var TestFeatures = flag.String("test-features", "",
 // NumClusters is the number of clusters to run tests on. Each cluster only has
 // a single test running on it at a given time. The number of clusters equals the
 // number of test threads which can run concurrently.
-var NumClusters = flag.Int("num-clusters", 1,
+var NumClusters = flag.Int("num-clusters", util.EnvInt("NUM_CLUSTERS", 1),
```
Comment (karlkfi) [inline e2e/flags.go:93]: do we want some kind of prefix for e2e env vars, like `E2E_`

Comment (karlkfi) [reply e2e/flags.go:93]: E2E_FLEET_SIZE would also be ok

Comment (sdowell) [reply e2e/flags.go:93]: yes dad

Comment (sdowell) [reply e2e/flags.go]: consider the E2E prefix slapped

**Review by karlkfi — APPROVED** [review]
/lgtm

**Pull request merged**